media/js/base/mozilla-client.js

```diff
@@ -0,0 +1,153 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Create namespace
+if (typeof Mozilla === 'undefined') {
+    var Mozilla = {};
+}
+
+(function() {
+    'use strict';
+
+    var Client = {};
+
+    /**
+     * Get the user's Firefox version number.
+     *
+     * @param  {String} ua - browser's user agent string, navigator.userAgent is used if not specified
+     * @return {String} version number.
+     */
+    Client._getFirefoxVersion = function (ua) {
+        ua = ua || navigator.userAgent;
+
+        var matches = /Firefox\/(\d+\.\d+(?:\.\d+)?)/.exec(ua);
+
+        return (matches !== null && matches.length > 0) ? matches[1] : '0';
+    };
+
+    /**
+     * Detect whether the user's Firefox is up to date or outdated. This data is mainly used for security notifications.
+     *
+     * @param  {Boolean} strict - whether the minor and patch-level version numbers should be compared. Default: true
+     * @param  {Boolean} isESR - whether the Firefox update channel is ESR
+     * @param  {String}  userVer - browser's version number
+     * @return {Boolean} result
+     */
+    Client._isFirefoxUpToDate = function (strict, isESR, userVer) {
+        strict = strict === undefined ? true : strict;
+        isESR = isESR === undefined ? window.isFirefoxESR() : isESR;
```

Comment:
Yes, the legacy `isFirefoxESR` function should be gone... [`navigator.buildID` will be removed from Firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=583181). There would be some false positives in case the API doesn't work, but `isFirefoxESR` also has false positives as noted, so it should be acceptable.